src/sentry/web/frontend/accounts.py

```diff
@@ -175,11 +175,15 @@ def account_settings(request):
 
         form.save()
 
-        # remove previously valid email address
+        # update notification settings for those set to primary email with new primary email
+        options = UserOption.objects.filter(user=user, value=old_email)
```

Comment:
I see now that we were doing this before. I'm not sure this is a great idea. We shouldn't be relying on querying based on the value here imo.